spark/src/main/java/com/robinhood/spark/SparkView.java

```diff
@@ -730,18 +743,38 @@ static int getNearestIndex(List<Float> points, float x) {
 
     @Override
     public void onScrubbed(float x, float y) {
+
+        // If x is within the bounds of an event, snap to that event.
+        @Nullable Pair<Integer, Float> eventInfo = getEventToSnapTo(x);
+        if (eventInfo != null) {
+            x = eventInfo.second;
+        }
+
         if (adapter == null || adapter.getCount() == 0) return;
         if (scrubListener != null) {
             getParent().requestDisallowInterceptTouchEvent(true);
-            int index = getNearestIndex(xPoints, x);
-            if (scrubListener != null) {
+            int index = eventInfo != null ? eventInfo.first : getNearestIndex(xPoints, x);
+            if (scrubListener != null && index != lastIndexSentToScrubListener) {
```

Comment:
This also fixes an existing bug where you could get notified multiple times about the same point in the data model if the point density of the graph was sparse enough (and snapping to events makes it much sparser around an event, which is how I noticed this)